commit 106b7a4eafc6d70047da0886438be9a328b3548e
Author: Tapas Kundu <tapas.kundu@broadcom.com>
Date:   Thu Jul 17 10:01:12 2025 +0000

    hardened sshd_config

    This change has hardened sshd_config as per
    STIG and also kex and ciphers which are not
    FIPS compliant are removed

    Signed-off-by: Tapas Kundu <tapas.kundu@broadcom.com>

diff --git a/sshd_config b/sshd_config
index 36894ace5..23ab41eb7 100644
--- a/sshd_config
+++ b/sshd_config
@@ -23,15 +23,15 @@
 #RekeyLimit default none
 
 # Logging
-#SyslogFacility AUTH
-#LogLevel INFO
+SyslogFacility AUTHPRIV
+LogLevel INFO
 
 # Authentication:
 
-#LoginGraceTime 2m
-#PermitRootLogin prohibit-password
-#StrictModes yes
-#MaxAuthTries 6
+LoginGraceTime 30
+PermitRootLogin no
+StrictModes yes
+MaxAuthTries 6
 #MaxSessions 10
 
 #PubkeyAuthentication yes
@@ -48,26 +48,26 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # For this to work you will also need host keys in /etc/ssh/ssh_known_hosts
 #HostbasedAuthentication no
 # Change to yes if you don't trust ~/.ssh/known_hosts for
-# HostbasedAuthentication
-#IgnoreUserKnownHosts no
+HostbasedAuthentication no
+IgnoreUserKnownHosts yes
 # Don't read the user's ~/.rhosts and ~/.shosts files
-#IgnoreRhosts yes
+IgnoreRhosts yes
 
 # To disable tunneled clear text passwords, change to no here!
 #PasswordAuthentication yes
-#PermitEmptyPasswords no
+PermitEmptyPasswords no
 
 # Change to no to disable s/key passwords
 #KbdInteractiveAuthentication yes
 
 # Kerberos options
-#KerberosAuthentication no
+KerberosAuthentication no
 #KerberosOrLocalPasswd yes
 #KerberosTicketCleanup yes
 #KerberosGetAFSToken no
 
 # GSSAPI options
-#GSSAPIAuthentication no
+GSSAPIAuthentication no
 #GSSAPICleanupCredentials yes
 
 # Set this to 'yes' to enable PAM authentication, account processing,
@@ -79,22 +79,22 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+UsePAM yes
 
-#AllowAgentForwarding yes
-#AllowTcpForwarding yes
+AllowAgentForwarding no
+AllowTcpForwarding no
 #GatewayPorts no
 #X11Forwarding no
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
 #PrintMotd yes
-#PrintLastLog yes
-#TCPKeepAlive yes
-#PermitUserEnvironment no
-#Compression delayed
-#ClientAliveInterval 0
-#ClientAliveCountMax 3
+PrintLastLog yes
+TCPKeepAlive no
+PermitUserEnvironment no
+Compression no
+ClientAliveInterval 900
+ClientAliveCountMax 1
 #UseDNS no
 #PidFile /var/run/sshd.pid
 #MaxStartups 10:30:100
@@ -103,14 +103,16 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #VersionAddendum none
 
 # no default banner path
-#Banner none
+Banner /etc/issue
 
 # override default of no subsystems
 Subsystem	sftp	/usr/libexec/sftp-server
 
 # Example of overriding settings on a per-user basis
 #Match User anoncvs
-#	X11Forwarding no
-#	AllowTcpForwarding no
+X11Forwarding no
 #	PermitTTY no
 #	ForceCommand cvs server
+KexAlgorithms ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
+Ciphers aes128-ctr,aes192-ctr,aes256-ctr,aes256-gcm@openssh.com,aes128-gcm@openssh.com
+MACs hmac-sha2-512,hmac-sha2-256
